### ISSUE TITLE
`apply_transformations` does not exist, but is referenced in docs

### ISSUE THREAD

**Issue by LFalch** [ggez/ggez#1018]
Documentation specifies to use `apply_transformations` after calling `mul_projection` or `set_projection`, but no such function exists.
Furthermore, I can't find anything in the changelog about this or the related `push_transform` and `pop_transform`. I can't really tell if this is merely a documentation problem; I have yet to figure out how to re-write my old code to use the new versions.

**Comment by PSteinhaus** [comment]
This could indeed be either a documentation problem, or a real one, though I sadly guess that it's the latter.

`set_projection` and `mul_projection` both just change `GraphicsContextGeneric::projection`, which only really comes into play when `GraphicsContextGeneric::set_global_mvp` is called, which, in turn, is done whenever a `SpriteBatch` or `MeshBatch` is drawn and also when `set_screen_coordinates` is called.

But, if you were to render, say, an `Image` all on its own, then your beautiful projection matrix wouldn't be used yet.
As a workaround you could just call `set_screen_coordinates` with the current screen coordinates after you called `set_projection` or `mul_projection`, but that's really just a workaround.

I guess `apply_transformations` was removed by accident and we have to re-add it. Either that, or we just call `set_global_mvp` at the end of `set_projection` and `mul_projection` ourselves internally and remove the note from the docs, but I guess that would be a bit more ugly, as people who supply their own matrices probably also don't mind calling `apply_transformations` whenever they're done themselves.

**Comment by LFalch** [comment]
I find out about this because my game used the old functions to offset non-hud drawing. I figured out I can achieve the same now by changing the screen coordinates instead. I expected it to be a deliberate removal, but I couldn't find anything detailing it. I don't know if I'm missing something somewhere. Re-adding it as `apply_projection` and making the documentation no longer have dead links seems good :)

In summary, I don't think you missed anything. My problems have been fixed and the documentation will no longer be wrong.
